content/troubleshooting/accessing-builder-machine-via-ssh.md

```diff
@@ -1,49 +1,64 @@
 ---
 description:
-  'Establish SSH access to the builder virtual machine to inspect your
-  build.  '
-title: Accessing virtual machine via SSH
+  'Establish SSH or VNC access to the virtual build machine'
+title: Remote access to build machine
 weight: 2
 ---
 
-Establishing SSH access to the virtual machine running the build is a secure way for inspecting your build during runtime and for a short time after the build has finished. Building in SSH mode gives you the following advantages:
+You can establish remote access to the virtual build machine running your build either via SSH or using a VNC client. 
 
-- See the processes running during the CI job
-- Reproduce all commands run during the build
-- Debug your custom scripts
+The **SSH access** allows you to access the build machine only through a terminal and run commands on it. This is a great option for debugging your builds. You can see the processes running during the CI job, reproduce all commands run during the build or debug your custom scripts.
 
-## How SSH access works
+The **VNC client** allows you to graphically access the remote build machine. It has multiple versions of Xcode and iOS simulators preinstalled, so you can run and test your iOS apps or change Xcode configuration files without owning a Mac yourself. 
 
-All virtual machines are located within the private network. In order to allow users to connect to the virtual machine, Codemagic creates a temporary SSH tunnel through the public gateway. A unique SSH key is generated before each build which is valid for the duration of the build. The SSH tunnel and the unique key are revoked after the build is finished and cannot be reused.
+The remote session can be started during runtime or shortly after the build has finished. 
```

Comment:
After the builder machine finish the build you not able to access the machine again, so i think you should remove this section "or shortly after the build has finished."